src/tables.jl

```diff
@@ -0,0 +1,87 @@
+# Tables and TableTraits interface
+
+## Chains
+
+Tables.istable(::Type{<:Chains}) = true
+
+Tables.columnaccess(::Type{<:Chains}) = true
+
+Tables.columns(chn::Chains) = chn
+
+Tables.columnnames(chn::Chains) = (:Iteration, :Chain, Symbol.(names(chn))...)
+
+function Tables.getcolumn(chn::Chains, i::Int)
+    return Tables.getcolumn(chn, Tables.columnnames(chn)[i])
+end
+function Tables.getcolumn(chn::Chains, nm::Symbol)
+    if nm == :Iteration
+        niter, _, nchains = size(chn)
+        return repeat(Base.OneTo(niter), nchains)
+    elseif nm == :Chain
+        chainids = chains(chn)
+        niter = size(chn, 1)
+        return vcat(fill.(chainids, niter)...)
+    else
+        return vec(getindex(chn, nm).value)
+    end
+end
+
+Tables.rowaccess(::Type{<:Chains}) = true
+
+Tables.rows(chn::Chains) = chn
+
+Tables.rowtable(chn::Chains) = Tables.rowtable(Tables.columntable(chn))
+
+function Tables.namedtupleiterator(chn::Chains)
+    return Tables.namedtupleiterator(Tables.columntable(chn))
+end
+
+function Tables.schema(chn::Chains)
+    nms = Tables.columnnames(chn)
+    T = eltype(chn.value)
+    types = (Int, Int, ntuple(_ -> T, size(chn, 2))...)
+    return Tables.Schema(nms, types)
+end
+
+IteratorInterfaceExtensions.isiterable(::Chains) = true
+function IteratorInterfaceExtensions.getiterator(chn::Chains)
+    return Tables.datavaluerows(Tables.columntable(chn))
+end
+
+TableTraits.isiterabletable(::Chains) = true
+
+## ChainDataFrame
+
+Tables.istable(::Type{<:ChainDataFrame}) = true
+
+Tables.columnaccess(::Type{<:ChainDataFrame}) = true
+
+Tables.columns(cdf::ChainDataFrame) = cdf
+
+Tables.columnnames(cdf::ChainDataFrame) = keys(cdf.nt)
+
+function Tables.getcolumn(cdf::ChainDataFrame, i::Int)
+    return Tables.getcolumn(cdf, keys(cdf.nt)[i])
+end
+Tables.getcolumn(cdf::ChainDataFrame, nm::Symbol) = getproperty(cdf.nt, nm)
```

Comment:
I'm curious, is there a reason for using `getproperty(cdf.nt, nm)` instead of `cdf.nt[nm]`?

Reply:
Not really, just preference. I usually use the dot syntax with named tuples instead of indexing. I can change if you prefer.

Reply:
I'm just a bit surprised about the use of `getproperty` with NamedTuples, I would have expected `getfield` rather (which is actually what `getindex` calls in https://github.com/JuliaLang/julia/blob/f449765943ba414bd57c3d1a44a73e5a0bb27534/base/namedtuple.jl#L113). So yes, I think either change it to `getfield` or use standard indexing.

Reply:
Ah, I did not know that. Sure I can do that.